addons/sale/wizard/sale_make_invoice_advance.py

```diff
@@ -9,6 +9,55 @@ class sale_advance_payment_inv(osv.osv_memory):
     _name = "sale.advance.payment.inv"
     _description = "Sales Advance Payment Invoice"
 
+    def default_get(self, cr, uid, fields, context=None):
+        res = super(sale_advance_payment_inv, self).default_get(cr, uid, fields, context=context)
+        vals = self._amount_total(cr, uid, context=context)
+        res.update(vals)
+        return res
+
+    def _get_amounts(self, cr, uid, ids, field_name, arg, context=None):
+        vals = self._amount_total(cr, uid, context=context)
+        return dict.fromkeys(ids, vals)
+
+    def _amount_calculate(self, invoice):
+        invoiced, paid = 0, 0
+        if invoice.state not in ('cancel', 'draft'):
+            invoiced = invoice.amount_total
+        if invoice.state == 'paid':
+            paid = invoice.amount_total
+        if invoice.state == 'open':
+            paid = invoice.amount_total - invoice.residual
+        return invoiced , paid
+
+    def _currency_position(self, amount, order):
+        amount_paid = (amount, order.currency_id.symbol)
+        if order.currency_id.position == 'before':
```

Comment:
I think, we should display currency field position using monetary widget from view side. so can you remove this stuff from here.